### ISSUE TITLE
Reduced servo precision in PWM IO

### ISSUE THREAD

**Issue by dtex** [Moddable-OpenSource/moddable#300]
Looking at [the IO servo example](https://github.com/Moddable-OpenSource/moddable/blob/public/examples/io/pwm/servo-sweep/main.js) I think I see a problem. With this implementation, the servo range is being mapped to just 84 steps at 56hz ([29, 113]).

We should be able to move the servo in much smaller increments but when writing to PWM with just a small fraction of the IO resolution (15% in your example) we can't do that. Not that we're immediately concerned with 8-bit devices, but if this design were applied to an 8-bit device, we'd only have 20 steps available resulting in minimum steps of ~9° on a 180° servo.

Servo positioning should be handled by writing a duty cycle in microseconds. 

Also, I'm curious why there is an 18ms period instead of the typical 20ms for servos?

**Comment by dtex** [comment]
Dug a little and I see that we'd need to include Servo.h from ESP8266/Arduino to get access to the writeMicroseconds method. I checked to see if I could configure an analog pin in such a way that I could make analogWrite work, but nope that won't work.

We can't include Servo.h in builtin/pwm since it's usually not needed, that would be wasteful when all we really need is writeMicroseconds (I'm making an assumption here correct me if I'm wrong). 

Here's the frustrating thing. The universe of platforms expect pwm in a variety of units:
* µs - [0, 1_000_000/frequency in hz] Requires that user's code factor for the frequency
* [0, 2 ** resolution] Common but loses precision and is a non-starter for servos
* servo degrees - [0, 180] Also a precision loss (and don't even get me started on my "not all servos are 180°" rant)
* [0, 1.0] - requires floating-point operations

I believe the correct unit for all PWM is µs because everything I found maps to µs before being converted to ticks. But even if we made the default unit for PWM be µs we still don't have a way to write that value on any Arduino based platform without Servo.h.

Bottom line, I think the only practical solution is to add builtin/servo and have that IO's write method expect µs. The thing that stinks is we might find that it's only strictly necessary for platforms whose API matches Arduino core. Other platforms may not need it.
